Source/common/MOLAuthenticatingURLSession.mm

```diff
@@ -264,6 +270,38 @@ - (NSURLCredential *)clientCredentialForProtectionSpace:(NSURLProtectionSpace *)
     if (certificate) CFRelease(certificate);
     if (clientCert) [self log:@"[Client Trust] Certificate: %@", clientCert];
 
+    // We have an identity but we don't know whether the private key is accessible to us, and if it
+    // isn't the framework will not give us any useful feedback. So, pull the private key from the
+    // identity and attempt to sign some random data with it. This is replicating what will happen
+    // during the mTLS handshake.
+    OSStatus status;
+    ScopedSecKeyRef scopedPrivateKey;
+    std::tie(status, scopedPrivateKey) = ScopedSecKeyRef::AssumeFrom(^OSStatus(SecKeyRef *out) {
+      return SecIdentityCopyPrivateKey(foundIdentity, out);
+    });
+    if (status != errSecSuccess) {
+      // This should never really happen if we've managed to find an identity.
+      [self log:@"[Client Trust] Failed to copy private key, authentication will likely fail: %d",
```

Comment:
Nit: "copy" could give the impression Santa is doing something unsafe with a private key.

```suggestion
      [self log:@"[Client Trust] Failed to access private key, authentication will likely fail: %d",
```